contracts/tests/gummyroll-test.ts

```diff
@@ -79,57 +82,56 @@ describe("gummyroll", () => {
     const ixs = [allocAccountIx];
     if (numLeaves > 0) {
       const root = Array.from(tree.root.map((x) => x));
-      const leaf = Array.from(leaves[numLeaves - 1]);
+      const leaf: number[] = Array.from(leaves[numLeaves - 1]);
       const proof = getProofOfLeaf(tree, numLeaves - 1).map((node) => {
-        return {
-          pubkey: new PublicKey(node.node),
-          isSigner: false,
-          isWritable: false,
-        };
+        return node.node
       });
 
-      ixs.push(
-        Gummyroll.instruction.initGummyrollWithRoot(
+      let initGummyRollWithRootInstr = createInitGummyrollWithRootWithProofInstruction(
+        {
+          merkleRoll: merkleRollKeypair.publicKey,
+          authority: payer.publicKey,
+          candyWrapper: CANDY_WRAPPER_PROGRAM_ID,
+        },
+        {
           maxDepth,
-          maxSize,
+          maxBufferSize: maxSize,
           root,
           leaf,
-          numLeaves - 1,
-          "https://arweave.net/<changelog_db_uri>",
-          "https://arweave.net/<metadata_db_id>",
-          {
-            accounts: {
-              merkleRoll: merkleRollKeypair.publicKey,
-              authority: payer.publicKey,
-              candyWrapper: CANDY_WRAPPER_PROGRAM_ID,
-            },
-            signers: [payer],
-            remainingAccounts: proof,
-          }
-        )
+          index: numLeaves-1,
+          changelogDbUri: "https://arweave.net/<changelog_db_uri>",
+          metadataDbUri: "https://arweave.net/<metadata_db_id>"
+        },
+        proof
+      );
+      ixs.push(
+        initGummyRollWithRootInstr
       );
     } else {
       ixs.push(
-        Gummyroll.instruction.initEmptyGummyroll(maxDepth, maxSize, {
-          accounts: {
+        createInitEmptyGummyrollInstruction(
+          {
             merkleRoll: merkleRollKeypair.publicKey,
             authority: payer.publicKey,
             candyWrapper: CANDY_WRAPPER_PROGRAM_ID,
           },
-          signers: [payer],
-        })
+          {
+            maxDepth,
+            maxBufferSize: maxSize
+          }
+        )
       );
     }
-    let txId = await execute(Gummyroll.provider, ixs, [
+    let txId = await execute(anchor.getProvider(), ixs, [
```

Comment:
I would prefer using a `provider` object created at the top of the file, alongside `connection`, if possible